toolkit/scripts/cherry_pick.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -e
+
+function help {
+    echo "Cherry-pick commit to a specific branch and create Github PR"
+    echo "Usage:"
+    echo "[MANDATORY] -r REPOSITORY -> name of the repository in the format <owner>/<repo-name>"
+    echo "[MANDATORY] -p PR_NUMBER -> number of the PR to cherry-pick commit from"
+    echo "[MANDATORY] -t TARGET_BRANCH -> target branch to cherry-pick commit to"
+}
+
+function collect_pr_info {
+    repo=$1
+    pr_number=$2
+
+    commit_hash=$(gh pr view $pr_number --repo $repo --json mergeCommit --jq '.mergeCommit.oid')
+    original_pr_title=$(gh pr view $pr_number --repo $repo --json title --jq '.title')
+}
+
+function cherry_pick {
+    commit_hash=$1
+    target_branch=$2
+    repo=$3
+    pr_number=$4
+    original_pr_title=$5
+
+    tmp_branch="cherry-pick-$target_branch-$commit_hash"
+
+    echo "Cherry picking commit ($commit_hash) to target branch $target_branch"
+
+    # reset the current working tree to clean state
+    git reset --hard
+    git clean -df
+    git checkout -- .
+
+    # create a temporary branch from target branch to perform cherry pick
+    git fetch --all
+    git checkout -b "$tmp_branch" origin/"$target_branch"
+
+    if ! git cherry-pick -x "$commit_hash"; then
+        echo "Cherry pick failed. Displaying conflicts below"
+        git diff --diff-filter=U
+        gh pr comment "$pr_number" \
+            --repo "$repo" \
+            --body "Cherry-pick failed for branch \`$target_branch\`. See run logs for more details: $RUN_URL"
+    else
+        echo "pushing to remote"
+        git push -u origin "$tmp_branch"
+        echo "done pushing to remote"
+        new_pr=$(gh pr create \
+            -B "$target_branch" \
+            -H "$tmp_branch" \
+            --repo "$repo" \
+            --title "[AUTO-CHERRY-PICK] $original_pr_title - branch $target_branch" \
+            --body "This is an auto-generated pull request to cherry pick commit $commit_hash to $target_branch. Original PR: #$pr_number")
+        gh pr comment "$pr_number" \
+            --repo "$repo" \
+            --body "Cherry-pick succeeded for branch \`$target_branch\`. See pull request #$new_pr"
```

Comment:
Just a suggestion: since we're using `gh pr comment` in two places, we could extract it into a function to not have to repeat the whole call twice and only pass the PR number, repo, and message to the function.

Reply:
it's not a too complex command so I think keeping this way helps with clarity

Reply:
Yeah, after a second look I agree.